x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/search/action/SearchStatusResponse.java

```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.core.search.action;
+
+/**
+ * An interface for status response of the stored or running async search
+ */
+public interface SearchStatusResponse {
```

Comment:
There is quite a bit of code repetition between AsyncStatusResponse and EqlStatusResponse, and between TransportEqlAsyncGetStatusAction and TransportGetAsyncStatusAction, any reason for not moving more shared things into SearchStatusResponse and AsyncTaskIndexService respectively?

Reply:
@imotov  Thank for the feedback, addressed in 4321efb4b62fc5a0fafdfc9c3e613f93a57a73bb.

I was not able to combine AsyncStatusResponse and EqlStatusResponse together because it would complicate the wire  serialization/deserialization code.  As we already have `AsyncStatusResponse` form 7.11, I did not want to change that part of the code.